src/I18n/Parser/PoFileParser.php

```diff
@@ -142,11 +142,10 @@ protected function _addMessage(array &$messages, array $item)
 
         $translation = stripcslashes($translation);
 
-        if ($context && (!isset($messages[$singular]) || is_array($messages[$singular]))) {
+        if ($context !== null) {
             $messages[$singular]['_context'][$context] = $translation;
-        }
-        if ($context === null) {
-            $messages[$singular] = $translation;
+        } else {
+            $messages[$singular]['_context'][''] = $translation;
```

Comment:
@markstory the issue still persist for me. Just tried it against the current master a6bf3f04b6499eb3aacb19ffcad9958e3aa8c962